exonum/src/node/consensus.rs

```diff
@@ -532,35 +532,18 @@ impl NodeHandler {
         }
     }
 
-    /// Handles raw transaction. Transaction is ignored if it is already known, otherwise it is
-    /// added to the transactions pool.
-    #[cfg_attr(feature = "flame_profile", flame)]
-    pub fn handle_tx(&mut self, msg: RawTransaction) {
-        //trace!("Handle transaction");
+    /// Checks if transaction is new and adds it into pool.
+    fn handle_tx_inner(&mut self, msg: RawTransaction) -> Result<(), String> {
         let hash = msg.hash();
-        let tx = {
-            let service_id = msg.service_id();
-            match self.blockchain.tx_from_raw(msg.clone()) {
-                Ok(tx) => tx,
-                Err(e) => {
-                    error!("{}, service_id={}", e.description(), service_id);
-                    return;
-                }
-            }
-        };
 
         profiler_span!("Make sure that it is new transaction", {
             let snapshot = self.blockchain.snapshot();
             if Schema::new(&snapshot).transactions().contains(&hash) {
-                return;
+                let err = format!("Received already committed transaction, hash {:?}", hash);
```

Comment:
1. Ok
2. For internal messages, I guess you're right. For public API calls, I think, duplicated requests are not OK. Maybe it will be better to leave this line as is, and just remove `error!()` call in `handle_tx()`?